### ISSUE TITLE
Enable tag filtering within MapGrpcHealthChecksService()

### ISSUE THREAD

**Issue by Eagle3386** [grpc/grpc-dotnet#1561]
### Is your feature request related to a problem? Please describe.
While `Add[Async]Check()` allows for specification of tags to then filter for them via `MapHealthChecks()`'s `HealthCheckOptions.Predicate` for HTTP GET requests, there's no such support via `MapGrpcHealthChecksService()`.

### Describe the solution you'd like
I'd like to filter for tags via something like that mentioned `Predicate` property mechanism in `MapGrpcHealthChecksService()`, too.

### Describe alternatives you've considered
- Implementing an overloaded method - not really an option.
- Using `GrpcServiceEndpointConventionBuilder`'s `Add()` method - failed to do so due to lack of documentation on how to use that convention builder correctly for my purpose.

### Additional context
I'd like to use that filtering mechanism to separate between an `AddDbContextCheck()` readiness probe & another one for a liveness probe.

**Comment by JamesNK** [comment]
I took a look at this. Does https://github.com/grpc/grpc-dotnet/pull/1565 let you achieve what you want?

**Comment by Eagle3386** [comment]
Absolutely! 😍 Thanks so much, James!

Just out of curiosity: why did you opt for
```csharp
if (_options.Filter != null && !_options.Filter(new HealthResultKey(entry.Key, entry.Value.Tags)))
```
instead of
```csharp
if (!_options?.Filter(new HealthResultKey(entry.Key, entry.Value.Tags)))
```
in [GrpcHealthChecksPublisher.cs](https://github.com/grpc/grpc-dotnet/blob/852a2140eb55e41cb148e615977762735c5fc6f8/src/Grpc.AspNetCore.HealthChecks/GrpcHealthChecksPublisher.cs#L41)?

**Comment by JamesNK** [comment]
I think you'd have to write it as:

```csharp
if (!(_options?.Filter(new HealthResultKey(entry.Key, entry.Value.Tags)) ?? false))
```

At which point it looks confusing. A separate null check is simpler.

**Comment by JamesNK** [comment]
Btw I rewrote the PR and made some major changes. It now supports mapping health results to service names. It makes it easier to aggregate multiple health checks together.

**Comment by Eagle3386** [comment]
> I think you'd have to write it as:
> 
> ```cs
> if (!(_options?.Filter(new HealthResultKey(entry.Key, entry.Value.Tags)) ?? false))
> ```

Of course, sorry about the forgotten part. My bad.

> At which point it looks confusing. A separate null check is simpler.

Then I'd like to suggest the `is not null` pattern instead as `!=` could be overwritten.

Regarding the rewrite, I'm glad for the changes. Thanks for that, too! 👍🏻

**Comment by Eagle3386** [comment]
@JamesNK Was this shipped with 2.42 yesterday or which release will contain it?

**Comment by JamesNK** [comment]
No
